crates/task-impls/src/transactions.rs

```diff
@@ -75,101 +61,21 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
-                                    {
-                                        consensus.metrics.outstanding_transactions.update(1);
-                                        consensus
-                                            .metrics
-                                            .outstanding_transactions_memory_size
-                                            .update(i64::try_from(size).unwrap_or_else(|e| {
-                                                warn!(
-                                                    "Conversion failed: {e}. Using the max value."
-                                                );
-                                                i64::MAX
-                                            }));
-                                    }
-                                }
-                            })
-                            .await;
-                    }
-                };
-
-                return None;
-            }
-            HotShotEvent::LeafDecided(leaf_chain) => {
-                let mut included_txns = HashSet::new();
-                let mut included_txn_size = 0;
-                let mut included_txn_count = 0;
-                for leaf in leaf_chain {
-                    if let Some(ref payload) = leaf.block_payload {
-                        for txn in
-                            payload.transaction_commitments(leaf.get_block_header().metadata())
-                        {
-                            included_txns.insert(txn);
-                        }
-                    }
-                }
-                let consensus = self.consensus.read().await;
-                let txns = self.transactions.cloned().await;
-
-                let _ = included_txns.iter().map(|hash| {
-                    if !txns.contains_key(hash) {
-                        self.seen_transactions.insert(*hash);
-                    }
-                });
-                drop(txns);
-                self.transactions
-                    .modify(|txns| {
-                        *txns = txns
-                            .drain()
-                            .filter(|(txn_hash, txn)| {
-                                if included_txns.contains(txn_hash) {
-                                    included_txn_count += 1;
-                                    included_txn_size +=
-                                        bincode_opts().serialized_size(txn).unwrap_or_default();
-                                    false
-                                } else {
-                                    true
-                                }
-                            })
-                            .collect();
+                self.api
+                    .send_event(Event {
+                        view_number: self.cur_view,
+                        event: EventType::Transactions {
+                            transactions: transactions.clone(),
+                        },
```

Comment:
I'm not sure what the expected behavior is but I think we might still want to track transactions locally and propose the from them in the case where there are not blocks.  We might have decided against that because the proposers has to pay a fee and our naive block would surely not recuperate the fee in most cases.  Worth a double check

Reply:
I would also be against it from complexity standpoint. We'd be adding an edge case here for when none of the builders (when we support multiple) proposed a block and there are transactions in our mempool, which will need to be maintained and tested and be configurable for node operator needs in case it's not profitable. I think it would be better served by simply running a separate builder of our own configured to include all transactions or somesuch.